server/routes/api/posts.js

```diff
@@ -0,0 +1,102 @@
+const Posts = require('../../models/forums/Post');
+
+
+module.exports = (app) => {
+
+    // API to add a Post to the database
+    // author title and description are required to be sent in request body  
+    app.post('/api/posts/add', function (req, res) {
+        // check if valid author is being sent
+        // make sure author is not blank
+        var author = req.body.author;
+        if (!author) {
+            return res.status(400).send({
+                success: false,
+                message: 'Error: Author cannot be blank.'
+            });
+        }
+        // make sure author(user) exists
+        User.find({
+            usn: author
+        }, (err, previousUsers) => {
+            if (err) {
+                return res.status(500).send({
+                    success: false,
+                    message: 'Error: Server find error'
+                });
+            } else if (previousUsers.length == 0) {
+                return res.status(409).send({
+                    success: false,
+                    message: 'Error: Account does not exist.'
+                });
+            }
+            // make sure title is not blank
+            var title = req.body.title;
+            if (!title) {
+                return res.status(400).send({
+                    success: false,
+                    message: 'Error: Title cannot be blank.'
+                });
+            }
+            // make sure description is not blank
+            var description = req.body.description;
+            if (!description) {
+                return res.status(400).send({
+                    success: false,
+                    message: 'Error: Description cannot be blank.'
+                });
+            }
+            // since all the data in the body is valid we add it to the database
+            var newPost = new Posts();
+            newPost.author = author;
+            newPost.title = title;
+            newPost.description = description;
+            newPost.save((err, post) => {
+                if (err) {
+                    return res.status(500).send({
+                        success: false,
+                        message: 'Error: Server error. Post not added.'
+                    });
+                }
+                console.log(newPost._id + "Post Added")
+                return res.status(200).send({
+                    success: true,
+                    message: 'Data Posted',
+                    post_id: newPost._id
+                });
+            });
+        });
+    });
+
+    // API to add a Comment to a post in the database
+    // Post ID, comment, author of comment are to be sent in request body
```

Comment:
So let's first write a route to add a comment. So like you have done above, you need to read the content of the comment from the body itself.

Reply:
So you don't need to look for ID.

Reply:
I added id because we need have some sort of identification for the post the comment is being added to, so i thought _id which i send back on successfully creating a post can be sent when a comment is added to that post, so we can do a findOneandUpdate operation

Reply:
Oh right. Makes sense. My bad. This seems fine.